addons/account/models/account_analytic_account.py

```diff
@@ -65,6 +65,39 @@ def _compute_vendor_bill_count(self):
         for account in self:
             account.vendor_bill_count = data.get(account.id, 0)
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_check_for_existing_analytic_distribution(self):
+        # check if the analytic line is linked to a record
+        # get all tables that has a column name called 'analytic_distribution'
+        self.env.cr.execute(
+            """
+                SELECT T.table_name
+                FROM information_schema.tables T
+                INNER JOIN information_schema.columns C ON C.table_name = T.table_name AND C.table_schema = T.table_schema
+                WHERE C.column_name LIKE 'analytic_distribution'
+                AND T.table_schema NOT IN ('information_schema', 'pg_catalog')
+                AND T.table_type = 'BASE TABLE'
+            """
+            )
+        all_tables_name = self.env.cr.fetchall()
+
+        for table_name in all_tables_name:
+            query = psycopg2.sql.SQL(
+                """
+                    SELECT COUNT(id)
+                    FROM {table_name}
+                    WHERE analytic_distribution->'%s' IS NOT NULL
```

Comment:
we could add a limit 1 here, as long as one record exists we can already throw an error. The count would be useless in that case as well